### DISCUSSION TITLE
Is there a way to make `Automatically download and install updates in the future` button selected by default?

### DISCUSSION THREAD

**Discussion by Tj3n** [sparkle-project/Sparkle#2208] [Q&A] [answered]
Hi, as the title stated, is there a way to make the `Automatically download and install updates in the future` button be selected by default when it appear in the `Software Update` popup? Thanks!

**Comment by zorgiepoo** [comment, marked as answer]
If you want your app to automatically download and install updates when possible, set `SUAutomaticallyUpdate` key in your app's Info.plist. More information in [Customizing Sparkle](https://sparkle-project.org/documentation/customization/).

**Reply by Tj3n** [reply]
> If you want your app to automatically download and install updates when possible, set `SUAutomaticallyUpdate` key in your app's Info.plist. More information in [Customizing Sparkle](https://sparkle-project.org/documentation/customization/).

Thank @zorgiepoo ! I'm quite aware of the `SUAutomaticallyUpdate`, just that the requirement is that we want to allow user to view the prompt normally on the first time, but with that button automatically checked (to increase the chance user do auto update from 2nd time), not sure if that's possible?

**Reply by zorgiepoo** [reply]
Sparkle intends users to be in control of changing options past the developer's initial configuration.. It is awkward to change the option underneath them and could be potentially confusing; users may not always pay much attention to that option on the alert window if your goal is awareness. If you change it for them, you also need to consider what to do if the user skips or dismisses the update then. It's probably possible to hack this on by changing the default before the alert shows up _only_ on the first time but I don't recommend it. By the way they can always get to the update alert and change the option if they initiate a check for updates themselves.

**Reply by Tj3n** [reply]
Thank you, I guess it's okay to change the default before the alert show up on the first time, atleast it does give user choice I think...
Would you be able to help tell if I can do it via any delegate function (when the update alert going to shown) or I will have to make changes directly in the code, thanks! I suppose I can change that by `[[NSUserDefaults standardUserDefaults] setBool:YES forKey:@"SUAutomaticallyUpdate"];`, but not sure where to put.

**Reply by zorgiepoo** [reply]
I think the proper way to do this is to add a checkbox option for automatically downloading updates to the update permission prompt that asks the user permission to check for new updates automatically.

Otherwise I can dig up an appropriate delegate method to use..

**Reply by Tj3n** [reply]
Thanks, hope you can tell what delegate method to use, we also have another checkbox in app own setting, but usually people won't bother look into that... currently I'm setting that in `- (BOOL)updaterShouldShowUpdateAlertForScheduledUpdate:(SUUpdater *)updater forItem:(SUAppcastItem *)item` (set everytime the Sparkle alert shown via background check)

**Reply by Tj3n** [reply]
Beside that, I have another question - is this 2 call equivalent since I see it does the same thing in the alert UI:
`[[NSUserDefaults standardUserDefaults] setBool:YES forKey:@"SUAutomaticallyUpdate"];`
and
`[[SUUpdater sharedUpdater] setAutomaticallyDownloadsUpdates: YES]`
Thanks!

**Reply by zorgiepoo** [reply]
[-[SPUStandardUserDriverDelegate standardUserDriverWillHandleShowingUpdate:forUpdate:state:]](https://sparkle-project.org/documentation/api-reference/Protocols/SPUStandardUserDriverDelegate.html#/c:objc(pl)SPUStandardUserDriverDelegate(im)standardUserDriverWillHandleShowingUpdate:forUpdate:state:) should be a better method. You ought to care about user initiated checks too, but if not then check the `state`.

Yes `-setAutomaticallyDownloadsUpdates:` just sets that default.

Note if we ever change Sparkle so that the option to automatically download/install updates in the future is in the permission prompt for update checks, this will break your workflow. But I think it is the right thing to do instead of what you're trying to do now.

**Reply by zorgiepoo** [reply]
https://github.com/sparkle-project/Sparkle/issues/2209

**Reply by Tj3n** [reply]
Got it, thanks! I agree with you, this is only temporary workflow for now to see if the adaptation rate raise or no, I'll be sure to check it in the future updates.

**Reply by zorgiepoo** [reply]
> Note if we ever change Sparkle so that the option to automatically download/install updates in the future is in the permission prompt for update checks, this will break your workflow. But I think it is the right thing to do instead of what you're trying to do now.

I put up a change to do this in #2285
